### PR TITLE
Support disabling autoscroll when hovering some elements

### PR DESCRIPTION
**Pull request by Slayer95**
Useful for in-chat GUIs, such as the one used by polls.

### CODE REVIEW

**Comment by Zarel** [comment]
Should we re-enable autoscroll when mousing out of said elements?

**Comment by Slayer95** [comment]
I guess... Why not?

File: js/client-chat.js [hunk 1]
```diff
@@ -1022,8 +1036,11 @@
 				}
 			}
 			if (userlist) this.addRow(userlist);
-			if (autoscroll) {
+			if (this.getAutoScroll()) {
 				this.$chatFrame.scrollTop(this.$chat.height());
+			} else if (!this.scrollHandler) {
+				this.scrollHandler = _.bind(ConsoleRoom.scrollHandler, this);
+				this.$chatFrame.on('scroll', this.scrollHandler);
```
Comment (Zarel) [inline js/client-chat.js]: Can you explain what this is doing?

Comment (Zarel) [inline js/client-chat.js]: The rest makes sense to me, but this is kind of confusing. Why are you attaching an extra scroll handler here?

Comment (Slayer95) [inline js/client-chat.js]: This is the only scroll handler, and it's intended to clear subtle notifications once the user scrolls down to the bottom of the chat log.

**Comment by Zarel** [comment]
> > Should we re-enable autoscroll when mousing out of said elements?
>
> I guess... Why not? o.o

Sorry, I think this was a misinterpretation. I mean, should we automatically scroll to the bottom when mousing out of said elements?

File: js/client-chat.js (hunk 1 above)
Comment (Zarel) [inline js/client-chat.js]: Oh, I see. Maybe add comments to the top of `scrollHandler` describing that behavior?

**Comment by Slayer95** [comment]
Oh. That would be handy for short scrolling distances.
For other circumstances, such behavior may block users' attemps to read old logs (which is a very annoying thing Facebook itself does).
Actually, even for "short distances" (<em>to be defined</em>), that would be problematic if moving the pointer upwards from a "no-autoscroll" area scrolled the chat log so that the pointer got back there. So maybe only if the user exited the element from the bottom?

File: js/client-chat.js (hunk 1 above)
Comment (Slayer95) [inline js/client-chat.js]: Fine

**Comment by Zarel** [comment]
> For other circumstances, such behavior may block users' attemps to read old logs (which is a very annoying thing Facebook itself does).

We should probably have a "There are new messages" bar at the bottom of chat, instead, in these cases.

> Actually, even for "short distances" (<em>to be defined</em>), that would be problematic if moving the pointer upwards from a "no-autoscroll" area scrolled the chat log so that the pointer got back there.

I don't consider it problematic. Scrolling up should be one step (no current need to animate it), and then whether or not you're in a no-autoscroll area again afterwards isn't exactly a huge problem.

**Comment by Slayer95** [comment]
> We should probably have a "There are new messages" bar at the bottom of chat, instead, in these cases.

I considered this initially, but then I thought we might as well use the subtle notifications system for this.

**Comment by Zarel** [comment]
Let's do both. At the very least, an orange line at the bottom of the chat log.

**Comment by Zarel** [comment]
poke?

**Comment by Slayer95** [comment]
On the radar!

**Comment by Zarel** [comment]
re-poke?

**Comment by Slayer95** [comment]
This doesn't work on mobile.

**Comment by Zarel** [comment]
I mean, it's still useful on desktop... :|

**Pull request closed without merging**